### PR TITLE
Fixing uses of sed in #8985 which do not work on MacOS X

### PR DESCRIPTION
**Pull request by herbelin**
**Kind:** compilation fix

I don't know if this is the best fix we can imagine but it works for me.

### CODE REVIEW

**Comment by ejgallego** [comment]
Note we already have https://github.com/coq/coq/pull/9085 which for some reason it has not been merged.

**Comment by herbelin** [comment]
Sorry, I did not know about #9085.

Unfortunately, I tested #9085 and it does not work for me. It complains with `command i expects \ followed by text`.

**Comment by ejgallego** [comment]
What OSX version are you using @herbelin ?

**Comment by ejgallego** [comment]
> What OSX version are you using @herbelin ?

AFAICT the sed invocation used there should work.

**Comment by herbelin** [comment]
I'm using 10.11 (El Capitan). I don't find a version number for `sed` though.

If I just do `sed -e "1i foo"` in a shell, it fails the same (`sed: 1: "1i foo
": command i expects \ followed by text`).

**Comment by ejgallego** [comment]
Wow, that's quite strange; I am not `sed` OSX expert so I guess I'll let an expert handle this.

**Comment by herbelin** [comment]
Without this PR I have to patch my archive for compiling. Can the first person who sees no problem with this PR merge it? This works for me. This works on CI (hence on linux I guess). Appveyor failure seems due to a timeout.

@mattam82 did not report anything and I'd be curious of why he does not meet the POSIX `sed` problem, but in any case, this is, formally speaking, "blocking" for me (and presumably for everyone working in a similar environement as mine).

**Comment by mattam82** [comment]
In my setup the sed fails but the compilation goes through. `sed` is not portable, for sure BSD sed and gnu sed have a different handling of -i.

**Comment by herbelin** [comment]
> In my setup the sed fails but the compilation goes through. 

For me, it goes through at the second time if I restart `make`. The first call to `make` fails (on a fresh archive). Is that the same for you?

**Comment by mattam82** [comment]
Nope, I did a first `git clean -dfx` and the compilation succeeded at first, using the standard make (no dune)

**Comment by herbelin** [comment]
With a `git clean -fxd` it seems indeed not to make the `make` failing (previously, I was doing `rm -r gramlib/.pack` and `make` was failing` afterwards).

In any case, the `sed` is failing and the `#` line is not added as expected. Is that what you observe too?

**Comment by mattam82** [comment]
Your version works for me. I'd like to merge but I'd like to get an ack from @ejgallego, @SkySkimmer  or @gares , it seems this issue was already discussed at length.

**Review by mattam82 — APPROVED** [review]

File: Makefile.build [hunk 1]
```diff
@@ -572,11 +572,11 @@ module Gramext = Gramlib__Gramext \
 module Grammar = Gramlib__Grammar" > $@
 
 gramlib/.pack/gramlib__P%: gramlib/p% | gramlib/.pack
-	cp -a $< $@
-	sed -e "1i # 1 \"$<\"" -i $@
+	printf "# 1 \"%s\"\n" $< > $@
```
Comment (SkySkimmer) [inline Makefile.build]:
```suggestion
	printf '# 1 "%s"\n' $< > $@
```

File: Makefile.build [hunk 2]
```diff
@@ -572,11 +572,11 @@ module Gramext = Gramlib__Gramext \
...
+	cat $< >> $@
 gramlib/.pack/gramlib__G%: gramlib/g% | gramlib/.pack
-	cp -a $< $@
-	sed -e "1i # 1 \"$<\"" -i $@
+	printf "# 1 \"%s\"\n" $< > $@
```
Comment (SkySkimmer) [inline Makefile.build]:
```suggestion
	printf '# 1 "%s"\n' $< > $@
```

**Comment by gares** [comment]
I'd recommend using `echo` (or `printf` as Gaetan suggests)

**Comment by mattam82** [comment]
@herbelin if you fix the quoting as suggested by Gaëtan I can merge quickly.

**Comment by herbelin** [comment]
Now using single quotes.

**Comment by mattam82** [comment]
Perfect, thanks!

**Pull request merged**